services/inputhost/pathCache.go

```diff
@@ -508,3 +536,26 @@ func (pathCache *inPathCache) getState() *admin.DestinationState {
 
 	return destState
 }
+
+func (pathCache *inPathCache) drainExtent(extUUID string, updateUUID string, drainWG *sync.WaitGroup) {
+	defer drainWG.Done()
+	pathCache.RLock()
+	defer pathCache.RUnlock()
```

Comment:
is it necessary to hold the lock while you are waiting for the connDrainWG ? Imagine an input host that's been asked to drain an extent (and at the same time assigned a new extent). When this happens, the client will try to reconnect to this input host (assuming the old conn is virtually closed in response to a DRAIN). But the Open call from the client cannot succeed because the pathCache lock is being held here.